substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -437,4 +438,36 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::AllowedRenewalDropped { core, when });
 		Ok(())
 	}
+
+	pub(crate) fn do_swap_leases(id: TaskId, other: TaskId) -> DispatchResult {
+		let mut id_exists = false;
+		let mut other_exists = false;
+		let leases = Leases::<T>::get()
+			.into_inner()
+			.into_iter()
+			.map(|mut lease| {
+				if lease.task == id {
+					lease.task = other;
```

Comment:
Good point though!

Reply:
I thought there could be multiple leases and we do want to swap all of them.

If this is the case, we'd better add another `ensure!` for it?

Reply:
https://github.com/paritytech/polkadot-sdk/pull/3714/commits/f5f6dce7d1276c365268659fd04b37947936cb87

Reply:
No Basti has a point. The "correctness" of the swap has already been verified on the relay chain, we should be very lenient on the Coretime chain. In particular, there could be a race and the lease of one para expired by the time the message arrives at the coretime chain (admittedly super unlikely edge case, but still), we should still do the swap, although one of the leases does not exist.

I mean that race in particular is really super niche, if a user updates that close he could miss the deadline himself already.

Nevertheless, I would err on as little ensures as possible: The swap is confirmed, it actually has to be good.

Reply:
Yes, I came to the same conclusion and removed everything. Even if someone decides to use the extrinsic directly it can be called again and 'undo' the result.